### ISSUE TITLE
The application does not load caches

### ISSUE THREAD

**Issue by Gabriele69** [evilbunny2008/geocaching#3]
After logging in and fixing the GPS, the application shows the blue arrow on the map, but does not display any caches

**Comment by Gabriele69** [comment]
Same problem even after the release of version 0.0.6. Can I help in some way to understand why it doesn't display the caches?

**Comment by ZikZakFR** [comment]
Same issue, even when giving the geocache ID nothing is displayed on the map.
UBports 16.04 on a Nexus 5 (hammerhead).

**Comment by vpzomtrrfrt** [comment]
Looks like it's breaking on date parsing.  The app expects year-month-day, but when I load the page, the dates look more like month/day/year

**Comment by vpzomtrrfrt** [comment]
I'm able to see one cache after the following patch, but it still errors out on something else:
```
diff --git a/py/util.py b/py/util.py
index 59c33f2..daa9d27 100644
--- a/py/util.py
+++ b/py/util.py
@@ -217,7 +217,10 @@ def clean_up(var):
     if var == "-":
         var = -1
     else:
-        var = time.mktime(datetime.datetime.strptime(var, "%Y-%m-%d").timetuple())
+        date_format = "%Y-%m-%d"
+        if "/" in var:
+            date_format = "%m/%d/%Y"
+        var = time.mktime(datetime.datetime.strptime(var, date_format).timetuple())
         var = int(var)
 
     return var
```

**Comment by vpzomtrrfrt** [comment]
Looks like my other issue is due to a "Premium-only geocache" being the second entry in the list.  For the date thing, I've sent PR #5.  Maybe that will fix your problem as well?

**Comment by Gabriele69** [comment]
> Looks like my other issue is due to a "Premium-only geocache" being the second entry in the list. For the date thing, I've sent PR #5. Maybe that will fix your problem as well?

Unfortunately I don't see a lot of activity from the developer ... I don't know how to program. Could you make a Fork by applying your modifications, so that we can download the package and verify that it works?

**Comment by seteahconnoc** [comment]
@Gabriele69 I've got the same problem that you mentioned in April. Any suggestions would be great.

bq Aquaris E5 HD, vegetahd, Ubuntu 16.04 (OTA-9), Geocaching for UT version 0.0.7

**Comment by evilbunny2008** [comment]
I just published v0.0.10, it should hopefully fix a bunch of mistakes I made

**Comment by seteahconnoc** [comment]
Hi, it 's me again. In the meantime, I got a new phone. Sadly, there are no caches shown. BTW: I'm a premium member at geocaching.com.

OnePlus One, bacon, Ubuntu 16.04 (OTA-12), Geocaching for UT version 0.0.12

**Comment by evilbunny2008** [comment]
Are there any errors when you tell the app to download cache list?

**Comment by seteahconnoc** [comment]
@evilbunny2008 Thanks for your quick response. In the app, the "loading wheel" is spinning once with no result. For further details on the behavior, I need to know, where to look for some kind of log.

**Comment by evilbunny2008** [comment]
I've misplaced my nexus5, but you debug apps by grabbing a copy of the clickable program (https://ubports.com/blog/ubports-blogs-news-1/post/introduction-to-clickable-147), you can then run clickable log or clickable logs to pull app output logs

Edit: I just tested the app on my laptop and it works fine, but without errors I'm stumped.

**Comment by seteahconnoc** [comment]
Thanks, so far. Just to be clear: In order to help me, you have to know some details. In order to get the details, I hava to run the command: `clickable logs` I guess, that this is regularly done by connecting my phone to an Ubuntu notebook. Is it possible to get the error log without connecting my phone to another computer? Is there a terminal command which I can use directly on my phone?

**Comment by evilbunny2008** [comment]
My pinephone turned up today updated to developer mode and ran the command line to turn on gps and the app failed to login, will get back to you when I have a fix

**Comment by seteahconnoc** [comment]
While learning about my Ubuntu phones (especially their connectivity), I'm looking forward for your insights. Thanks for your efforts in improving the geocaching app.

**Comment by evilbunny2008** [comment]
My problem seems to be because of a problem in python3 sqlite library for aarch64.

As for getting errors from terminal, you can cat the log file:
`cat ./.cache/upstart/application-click-geocaching.evilbunny_geocaching_0.0.12.log`

**Comment by seteahconnoc** [comment]
Thank you for telling me where to find the log file. Here is the main part of it 's information. (I left out information regarding my location.)

```
Attempting to reload the previous session.
Session re-loaded.
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.153\" (uid=32011 pid=30889 comm=\"/usr/lib/qt5/bin/qmlscene qml/Main.qml \") interface=\"com.canonical.libertine.Service.Operations\" member=\"list\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=3369 comm=\"/usr/bin/python3 /usr/bin/libertined \")"

...

User logged in, skipping re-auth
propsReply "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.159\" (uid=32011 pid=30889 comm=\"/usr/lib/qt5/bin/qmlscene qml/Main.qml \") interface=\"org.freedesktop.DBus.Properties\" member=\"GetAll\" error name=\"(unset)\" requested_reply=\"0\" destination=\"org.freedesktop.NetworkManager\" (uid=0 pid=1761 comm=\"NetworkManager \")"

...

Found cacheid: GC...
459 - Failed to parse cache info.
could not convert string to float: '1,5'

```

The last content (`Failed to parse cache info`) seems to be about the decimal separator (comma instead of point).

Hopefully, this might be helpful.

**Comment by seteahconnoc** [comment]
- At first, I thought about the messages "An AppArmor policy prevents this sender from sending this message to this recipient." - I still don't know how/if the app's functionality is affected by this.

- I opened my user settings on `www.geocaching.com`, and I set the language to "english".

- After stopping and restarting the app, I got to see the loaded caches.

So for me, it's "problem solved".

@evilbunny2008 Thanks again for the app and for the information on the log file's location.
